connectors/migrations/20241212_backfill_github_data_source_folders.ts

```diff
@@ -0,0 +1,212 @@
+import { getGithubCodeDirectoryParentIds } from "@connectors/connectors/github/lib/hierarchy";
+import { dataSourceConfigFromConnector } from "@connectors/lib/api/data_source_config";
+import { upsertFolderNode } from "@connectors/lib/data_sources";
+import {
+  GithubCodeDirectory,
+  GithubCodeFile,
+  GithubCodeRepository,
+  GithubDiscussion,
+  GithubIssue,
+} from "@connectors/lib/models/github";
+import { ConnectorModel } from "@connectors/resources/storage/models/connector_model";
+import type { DataSourceConfig } from "@connectors/types/data_source_config";
+
+async function main() {
+  const connectors = await ConnectorModel.findAll({
+    where: {
+      type: "github",
+    },
+  });
+
+  for (const connector of connectors) {
+    const dataSourceConfig = dataSourceConfigFromConnector(connector);
+    const repositories = await GithubCodeRepository.findAll({
+      where: {
+        connectorId: connector.id,
+      },
+    });
+
+    for (const repository of repositories) {
+      // upsert repository as folder
+      // Throws if error
+      const repoFolderId = await upsertRepositoryFolderNode(
+        repository,
+        dataSourceConfig
+      );
+
+      // Upsert Code folder if we have some (file or directory)
+      const shouldCreateCodeFolder = await repositoryContainsCode(repository);
+      if (shouldCreateCodeFolder) {
+        await upsertCodeFolderNode(
+          repository.repoId,
+          repoFolderId,
+          dataSourceConfig
+        );
+      }
+
+      const directories = await GithubCodeDirectory.findAll({
+        where: {
+          connectorId: connector.id,
+        },
+      });
+      // upsert directories as folders, in chunks
+      for (let i = 0; i < directories.length; i += 16) {
+        const chunk = directories.slice(i, i + 16);
+        await Promise.all(
+          chunk.map(async (directory) => {
+            await upsertDirectoryFolderNode(
+              directory,
+              repository.id,
+              dataSourceConfig
+            );
+          })
+        );
+      }
+
+      // Upsert issue folder if we have issues
+      const shouldCreateIssueFolder =
```

Comment:
is that inside the loop on repositories?
if so, IMO we need to move it outside because if you sync issues but not code you don't have a GithubCodeRepository created I think?